pkg/apis/workflow/v1alpha1/container_set_template_types.go

```diff
@@ -2,13 +2,49 @@ package v1alpha1
 
 import (
 	"fmt"
+	"time"
 
 	corev1 "k8s.io/api/core/v1"
+	intstr "k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 type ContainerSetTemplate struct {
-	Containers   []ContainerNode      `json:"containers" protobuf:"bytes,4,rep,name=containers"`
-	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty" protobuf:"bytes,3,rep,name=volumeMounts"`
+	Containers    []ContainerNode            `json:"containers" protobuf:"bytes,4,rep,name=containers"`
+	VolumeMounts  []corev1.VolumeMount       `json:"volumeMounts,omitempty" protobuf:"bytes,3,rep,name=volumeMounts"`
+	RetryStrategy *ContainerSetRetryStrategy `json:"retryStrategy,omitempty" protobuf:"bytes,5,opt,name=retryStrategy"`
```

Comment:
I think we could do with a comment for users here as this gets surfaced in docs. How does the retry work for example?

Reply:
Will update with docs.